### ISSUE TITLE
Problem when merging results of binned dataframes

### ISSUE THREAD

**Issue by benkrikler** [benkrikler/fast-carpenter-github-test#6]
**Imported from [gitlab issue 6](https://gitlab.cern.ch/fast-hep/public/fast-carpenter/issues/6)**

I think there's an issue merging binned dataframe outputs, possibly caused by the dataframe being empty for some reason.
```
Traceback (most recent call last):
  File "/users/bk17414/.local/bin/fast_carpenter", line 11, in <module>
    load_entry_point('fast-carpenter', 'console_scripts', 'fast_carpenter')()
  File "/software/bk17414/CHIP/fast-carpenter/fast_carpenter/__main__.py", line 69, in main
    return process.run(datasets, sequence)
  File "/users/bk17414/.local/lib/python2.7/site-packages/atuproot/atuproot_main.py", line 56, in run
    return self._run(loop)
  File "/users/bk17414/.local/lib/python2.7/site-packages/atuproot/atuproot_main.py", line 116, in _run
    result = loop()
  File "/users/bk17414/.local/lib/python2.7/site-packages/alphatwirl/datasetloop/loop.py", line 31, in __call__
    return self.reader.end()
  File "/users/bk17414/.local/lib/python2.7/site-packages/alphatwirl/datasetloop/reader.py", line 30, in end
    return [reader.end() for reader in self.readers]
  File "/users/bk17414/.local/lib/python2.7/site-packages/alphatwirl/loop/EventDatasetReader.py", line 89, in end
    self._merge(runid, reader)
  File "/users/bk17414/.local/lib/python2.7/site-packages/alphatwirl/loop/EventDatasetReader.py", line 101, in _merge
    merge_in_order(runid_reader_map, runid, reader)
  File "/users/bk17414/.local/lib/python2.7/site-packages/alphatwirl/loop/merge.py", line 30, in merge_in_order
    data.merge(data_next)
  File "/users/bk17414/.local/lib/python2.7/site-packages/alphatwirl/loop/ReaderComposite.py", line 56, in merge
    r.merge(o)
  File "/software/bk17414/CHIP/fast-carpenter/fast_carpenter/summary.py", line 85, in merge
    self.contents = self.contents.add(rhs.contents, fill_value=0)
  File "/users/bk17414/.local/lib/python2.7/site-packages/pandas/core/ops.py", line 1559, in f
    return self._combine_const(other, na_op, try_cast=True)
  File "/users/bk17414/.local/lib/python2.7/site-packages/pandas/core/frame.py", line 4779, in _combine_const
    try_cast=try_cast)
  File "/users/bk17414/.local/lib/python2.7/site-packages/pandas/core/internals.py", line 3687, in eval
    return self.apply('eval', **kwargs)
  File "/users/bk17414/.local/lib/python2.7/site-packages/pandas/core/internals.py", line 3581, in apply
    applied = getattr(b, f)(**kwargs)
  File "/users/bk17414/.local/lib/python2.7/site-packages/pandas/core/internals.py", line 1437, in eval
    'with block values'.format(other=other))
TypeError: Could not compare [None] with block values

```

**Comment by benkrikler** [comment]
**On 2018-11-14 Benjamin Krikler (bkrikler) wrote:**

mentioned in commit dcb58e7340f654cc1f8575c3a6a19c461376e2e9

**Comment by benkrikler** [comment]
**On 2018-11-14 Benjamin Krikler (bkrikler) wrote:**

mentioned in merge request [gitlab:!9](https://gitlab.cern.ch/fast-hep/public/fast-carpenter/merge_requests/9)

**Comment by benkrikler** [comment]
**On 2018-11-14 Benjamin Krikler (bkrikler) wrote:**

mentioned in commit 4bb103aa95f6166ddbe414968c037b97dbc4968e

**Comment by benkrikler** [comment]
**On 2018-11-14 Benjamin Krikler (bkrikler) wrote:**

closed via commit dcb58e7340f654cc1f8575c3a6a19c461376e2e9

**Comment by benkrikler** [comment]
**On 2018-11-14 Benjamin Krikler (bkrikler) wrote:**

closed via merge request [gitlab:!9](https://gitlab.cern.ch/fast-hep/public/fast-carpenter/merge_requests/9)
